src/toil/realtimeLogger.py

```diff
@@ -0,0 +1,190 @@
+            length = struct.unpack(">L", length_data)[0]
+            
+            # This is where we'll put the received message
+            message_parts = []
+            length_received = 0
+            while length_received < length:
+                # Keep trying to get enough data
+                part = self.rfile.read(length - length_received)
+                
+                length_received += len(part)
+                message_parts.append(part)
+                
+            # Stitch it all together
+            message = "".join(message_parts)
+
+            try:
+            
+                # Parse it as JSON
+                message_attrs = json.loads(message)
+                
+                # Fluff it up into a proper logging record
+                record = logging.makeLogRecord(message_attrs)
+            except:
+                logging.error("Malformed record")
+                
+            # TODO: do log level filtering
+            logging.getLogger("remote").handle(record)
+            
+class JSONDatagramHandler(logging.handlers.DatagramHandler):
+    """
+    Send logging records over UDP serialized as JSON.
+    """
+    
+    def makePickle(self, record):
+        """
+        Actually, encode the record as length-prefixed JSON instead.
+        """
+        
+        json_string = json.dumps(record.__dict__)
+        length = struct.pack(">L", len(json_string))
+        
+        return length + json_string
+        
+class RealtimeLogger(object):
+    """
+    All-static class for getting a logger that logs over UDP to the master.
+    
+    Usage:
+    
+    1. Make sure Job.Runner.startToil() is running on your master.
+    
+    2. From a running job on a worker, do:
+    
+    from toil.realtimeLogger import RealtimeLogger
+    realtime = RealtimeLogger.get()
+    realtime.info("This is a logging message that goes straight to the master")
+    
+    """
+    
+    # Also the logger
+    logger = None
+    
+    # The master keeps a server and thread
+    logging_server = None
+    server_thread = None
+  
+    @classmethod
+    def start_master(cls):
+        """
+        Start up the master server and put its details into the options
+        namespace.
+        
+        Python logging should have already been configured.
+        
+        """
+        
+        # Start up the logging server
+        cls.logging_server = SocketServer.ThreadingUDPServer(("0.0.0.0", 0),
+            LoggingDatagramHandler)
+            
+        # Set up a thread to do all the serving in the background and exit when
+        # we do
+        cls.server_thread = threading.Thread(
+            target=cls.logging_server.serve_forever)
+        cls.server_thread.daemon = True
+        cls.server_thread.start()
+        
+        # Set options for logging in the class and the options namespace
+        # Save them in the environment so they get sent out to jobs
+        os.environ["RT_LOGGING_HOST"] = socket.getfqdn()
```

Comment:
Every environment variable introduced by Toil should be prefixed with `TOIL_`. So I think `TOIL_RT_LOG_HOST` would be more appropriate.